lib/BC/SleighLifter.cpp

```diff
@@ -1263,6 +1267,27 @@ class SleighLifter::PcodeToLLVMEmitIntoBlock {
                                                   vars[2]);
         return kLiftedInstruction;
       }
+      if (other_func_name == kSysCallName &&
+          insn.arch_name == ArchName::kArchPPC) {
+        DLOG(INFO) << "Invoking syscall";
+
+        const auto [mem_ptr_ref, mem_ptr_ref_type] =
+            insn_lifter_parent.LoadRegAddress(
+                bldr.GetInsertBlock(), state_pointer, kMemoryVariableName);
+
+        // Get a LLVM value for the sync hyper call enumeration.
+        auto hyper_call_int =
+            static_cast<uint32_t>(SyncHyperCall::Name::kPPCSysCall);
+        auto hyper_call = llvm::ConstantInt::get(
+            llvm::IntegerType::get(this->context, 32), hyper_call_int);
+        std::array<llvm::Value *, 3> args = {state_pointer, mem_ptr_ref,
+                                             hyper_call};
+
+        bldr.CreateCall(insn_lifter_parent.GetIntrinsicTable()->sync_hyper_call,
```

Comment:
So is the reasoning here for why we need a custom signature for this sync call because we want to pass the type sync call explicitly? Just wondering why we cant take the strategy of other intrinsic hyper calls where it's a typical (state,PC,memory) lifted function and we can use AddCall transparently.

Reply:
Yep that's correct. With `__remill_sync_hyper_call`, we need to provide a `Name` enumeration to signal what kind of sync call we're making.